subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -132,92 +24,117 @@ static void strncpy_nullterm(char *dst, const char *src, size_t maxlen)
+							  "status");
+	if (cJSON_IsString(status) && status->valuestring != NULL) {
+		strncpy_nullterm(status_buf, status->valuestring,
+				 STATUS_MAX_LEN);
 	} else {
+		cJSON_Delete(update_response);
 		return -ENODATA;
 	}
 
+	cJSON_Delete(update_response);
 	return 0;
 }
 
-int aws_fota_parse_DescribeJobExecution_rsp(char *job_document,
+int aws_fota_parse_DescribeJobExecution_rsp(const char *job_document,
 					   u32_t payload_len,
 					   char *job_id_buf,
 					   char *hostname_buf,
 					   char *file_path_buf,
-					   int *version_number)
+					   int *execution_version_number)
 {
 	if (job_document == NULL
 	    || job_id_buf == NULL
 	    || hostname_buf == NULL
 	    || file_path_buf == NULL
-	    || version_number == NULL) {
+	    || execution_version_number == NULL) {
 		return -EINVAL;
 	}
 
-	struct desc_job_execution_obj root_obj;
-	struct job_document_obj *job_doc_obj;
-	struct execution_obj *exec_obj;
+	int ret;
 
-	int ret = json_obj_parse(job_document, payload_len,
-				 desc_job_execution_obj_descr,
-				 ARRAY_SIZE(desc_job_execution_obj_descr),
-				 &root_obj);
+	cJSON *json_data = cJSON_Parse(job_document);
 
-	if (ret < 0) {
-		return ret;
+	if (json_data == NULL) {
+		ret = -ENODATA;
+		goto cleanup;
 	}
 
-	if (!(ret & BIT(EXECUTION_OBJ_DECODED_BIT))) {
-		/* No execution object in JSON */
-		return 0;
+	cJSON *execution = cJSON_GetObjectItemCaseSensitive(json_data,
+							    "execution");
+	if (execution == NULL) {
+		ret = 0;
+		goto cleanup;
 	}
 
-	/* Check if the execution field of the object has been parsed. This
-	 * test is currently non-functional as Zephyr's JSON parser does not
-	 * populate fields as NULL in nested structure if the field in the
-	 * nested structure is not found.
-	 */
-	exec_obj = &root_obj.execution;
-	job_doc_obj = &exec_obj->job_document;
-	if (exec_obj->job_id != 0
-	    && exec_obj->version_number != 0
-	    && job_doc_obj->location.host != 0
-	    && job_doc_obj->location.path != 0) {
-		strncpy_nullterm(job_id_buf, exec_obj->job_id,
+	cJSON *job_id = cJSON_GetObjectItemCaseSensitive(execution, "jobId");
+
+	if (cJSON_IsString(job_id) && job_id->valuestring != NULL) {
+		strncpy_nullterm(job_id_buf, job_id->valuestring,
 				AWS_JOBS_JOB_ID_MAX_LEN);
-		*version_number = exec_obj->version_number;
-		strncpy_nullterm(hostname_buf,
-				job_doc_obj->location.host,
+	} else {
+		ret = -ENODATA;
+		goto cleanup;
+	}
+
+	cJSON *job_data = cJSON_GetObjectItemCaseSensitive(execution,
+							   "jobDocument");
+
+	if (!cJSON_IsObject(job_data)) {
+		ret = -ENODATA;
+		goto cleanup;
+	}
+
+	cJSON *location = cJSON_GetObjectItemCaseSensitive(job_data,
+							    "location");
+
+	if (!cJSON_IsObject(location)) {
+		ret = -ENODATA;
+		goto cleanup;
+	}
+
+	cJSON *hostname = cJSON_GetObjectItemCaseSensitive(location, "host");
+	cJSON *path = cJSON_GetObjectItemCaseSensitive(location, "path");
+
+	if ((cJSON_IsString(hostname) && hostname->valuestring != NULL)
```

Comment:
`cJSON_GetStringValue(hostname)`?

Reply:
I guess `cJSON_GetStringValue(hostname) != NULL` should suffice yes.